File: server/PCTeleLog/src/pctelelog/events/EventType.java
package pctelelog.events;

import org.codehaus.jackson.annotate.JsonIgnoreType;

@JsonIgnoreType
public enum EventType {
	/**
	 * Sent by the Client to notify connection
	 */
	CLIENT_CONNECT,
	
	/**
	 * A missed call
	 */
	MISSED_CALL,
	
	/**
	 * An incoming call, ie: The phone is ringing.
	 */
	INCOMING_CALL,
	
	/**
	 * A (incoming) call has ended
	 */
	CALL_ENDED,
	
	/**
	 * Client is disconnecting. 
	 */
	SHUTDOWN,
	
	/* Internal App or Protocol Events */
	/**
	 * Handshake event. Shouldn't be handled
	 */
	HELLO,
	
	/**
	 * Fired by the Java server in instances that it has to shutdown.
	 * This event would likely require the GUI to have to restart.
	 */
	SERVERSHUTDOWN, // Fired to EventListeners when server is shutting down
	
	/**
	 * Fired when a socket has closed (possibly unexpectedly)
	 */
	CLIENT_SOCKET_CLOSE // Fired when a client socket closes
}

File: pyqt-ui/src/pytelelog_pyqt/EventType.py

class EventType(object):
    
    '''
    The counterpart for EventType.java enum
    '''
    
    def __init__(self, gateway):
        self.INCOMING_CALL = gateway.jvm.pctelelog.events.EventType.INCOMING_CALL
        self.MISSED_CALL = gateway.jvm.pctelelog.events.EventType.MISSED_CALL
        self.CALL_ENDED = gateway.jvm.pctelelog.events.EventType.CALL_ENDED
        self.CLIENT_CONNECT = gateway.jvm.pctelelog.events.EventType.CLIENT_CONNECT
        self.SHUTDOWN = gateway.jvm.pctelelog.events.EventType.SHUTDOWN